packages/ui-components/src/Common/Select/StatelessSelect/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { ChevronDownIcon } from '@heroicons/react/24/solid';
+import classNames from 'classnames';
+import { useId, useMemo } from 'react';
+
+import type { SelectGroup, SelectProps } from '#ui/Common/Select';
+import type { LinkLike } from '#ui/types';
+import { isStringArray, isValuesArray } from '#ui/util/array';
+
+import styles from '../index.module.css';
+
+type StatelessSelectConfig = {
+  as?: LinkLike | 'div';
+};
+
+export type StatelessSelectProps<T extends string> = SelectProps<T> &
+  StatelessSelectConfig;
+
+const StatelessSelect = <T extends string>({
+  values = [],
+  defaultValue,
+  placeholder,
+  label,
+  inline,
+  className,
+  ariaLabel,
+  disabled = false,
+  as: Component = 'div',
+}: StatelessSelectProps<T>) => {
+  const id = useId();
+
+  const mappedValues = useMemo(() => {
+    let mappedValues = values;
+
+    if (isStringArray(mappedValues)) {
+      mappedValues = mappedValues.map(value => ({
+        label: value,
+        value: value,
+      }));
+    }
+
+    if (isValuesArray(mappedValues)) {
+      return [{ items: mappedValues }];
+    }
+
+    return mappedValues as Array<SelectGroup<T>>;
+  }, [values]) as Array<SelectGroup<T>>;
+
+  // Find the current/default item to display in summary
+  const currentItem = useMemo(
+    () =>
+      mappedValues
+        .flatMap(({ items }) => items)
+        .find(item => item.value === defaultValue),
+    [mappedValues, defaultValue]
+  );
+
+  return (
+    <div
+      className={classNames(
+        styles.select,
+        styles.noscript,
+        { [styles.inline]: inline },
+        className
+      )}
+    >
+      {label && (
+        <label className={styles.label} htmlFor={id}>
+          {label}
+        </label>
+      )}
+
+      <details className={styles.trigger} id={id}>
+        <summary
+          className={styles.summary}
+          aria-label={ariaLabel}
+          aria-disabled={disabled}
+        >
+          {currentItem ? (
```

Comment:
```suggestion
          {currentItem && (
```

And then for the placeholder:

```tsx
          {currentItem || (
```